code/ui/src/tabs/common/Setup.vue

```diff
@@ -0,0 +1,233 @@
+                                              type="text"
+                                              action="reconnect"
+                                              placeholder="255.255.255.0"
+                                              maxlength="15"
+                                              tabindex="0"
+                                              autocomplete="false"/>
+                                        <Hint>Usually 255.255.255.0 for /24 networks</Hint>
+                                    </C>
+
+                                    <C><label>DNS IP</label></C>
+                                    <C>
+                                        <Inpt name="dns"
+                                              type="text"
+                                              action="reconnect"
+                                              value="8.8.8.8"
+                                              maxlength="15"
+                                              tabindex="0"
+                                              autocomplete="false"/>
+                                        <Hint>
+                                            Set the Domain Name Server IP to use when using a static IP
+                                        </Hint>
+                                    </C>
+                                </template>
+                            </Row>
+                        </template>
+                        <template #btnRemove="tpl">
+                            <Btn name="del-network" color="danger" @click="tpl.click">Delete network</Btn>
+                        </template>
+                        <template #btnAdd="tpl">
+                            <Btn name="add-network" @click="tpl.click">Add network</Btn>
+                        </template>
+                    </Repeater>
+
+                    <Row>
+                        <C class="center" stretch :size="4" :offset="3">
+                            <Btn name="save"
+                                 title="Save">
+                                Save
+                            </Btn>
+                        </C>
+                    </Row>
+                </fieldset>
+                <fieldset>
+                    <legend>Or restore the settings from a backup</legend>
+                    <input type="text" :value="restoreFile.name" readonly @click="() => $refs.restoreFile.$el.click()">
+                    <Btn name="settings-restore" color="primary" @click="() => $refs.restoreFile.el.click()">
+                        Restore
+                    </Btn>
+                    <Inpt ref="restoreFile" name="restore" type="file" @change="(file) => { restoreFile = file }"/>
+                </fieldset>
+            </div>
+        </div>
+    </Form>
+</template>
+
+<script>
+    import Btn from "../../components/Button";
+    import Inpt from "../../components/Input";
+    import Hint from "../../components/Hint";
+    import C from "../../layout/Col";
```

Comment:
(nit) Not specifically related to Setup code, just in regard to naming
Any reason not to use full name like C -> Column, Btn -> Button?

Reply:
+1